framework/src/org/checkerframework/framework/type/AnnotatedTypeLoader.java

```diff
@@ -0,0 +1,372 @@
+        annotatedTypeQualSet.addAll(loadedAnnotations.keySet());
+
+        if (!excludePolyAll) {
+            // add PolyAll to the qualifier set unless it is explicitly excluded
+            annotatedTypeQualSet.add(PolyAll.class);
+        }
+
+        // return the set of annotation classes
+        return Collections.unmodifiableSet(annotatedTypeQualSet);
+    }
+
+    /**
+     * gets the set of annotation mirrors from the loaded annotations
+     * @return a hashset of the annotation mirrors
+     */
+    public final Set<AnnotationMirror> getAnnotationMirrorSet() {
+        return new HashSet<AnnotationMirror>(loadedAnnotations.values());
+    }
+
+    // loads annotations via reflection
+    private final Map<Class<? extends Annotation>, AnnotationMirror> loadCheckerFrameworkAnnotations() {
+        Map<Class<? extends Annotation>, AnnotationMirror> annos = new HashMap<Class<? extends Annotation>, AnnotationMirror>();
+
+        Set<String> annoFiles = getAnnotationNames();
+
+        for (String fileName : annoFiles) {
+            try {
+                String annoName = packageName + DOT + fileName;
+
+                // Load in the class files
+                Class<?> cls = Class.forName(annoName);
+
+                // ensure that the freshly loaded class is an annotation, and has the @Target annotation
+                if (cls.isAnnotation() && cls.getAnnotation(Target.class) != null) {
+                    // scan through the @Target annotation for its values
+                    for (ElementType element : cls.getAnnotation(Target.class).value()) {
+                        // ensure that the @Target annotation has the value of ElementType.TYPE_USE or ElementType.TYPE_PARAMETER
+                        if (AnnotatedTypes.hasTypeTarget(element)) {
+                            // if so, process the annotation and get its equivalent AnnotationMirror
+                            Class<? extends Annotation> annoClass = cls.asSubclass(Annotation.class);
+                            // createAnnotationMirrorFromClass is optionally overridden by each individual checker
+                            // returns an annotation mirror if the checker handles it, or 
+                            // null if it either doesn't handle it or fails to produce an annotation mirror
+                            AnnotationMirror convertedMirror = createAnnotationMirrorFromClass(annoClass);
+
+                            if (convertedMirror != null) {
+                                // convert the annotation class into an annotation mirror, add the class and the mirror into the hashmap
+                                annos.put(annoClass, convertedMirror);
+                            }
+
+                            break;
+                        }
+                    }
+                }
+            } catch (ClassNotFoundException e) {
+                //TODO: give better feedback
+                e.printStackTrace();
```

Comment:
Issue an error like:
ErrorReporter.errorAbort("AnnotatedTypeLoader: class not found: "+classname);